### DISCUSSION TITLE
I can't install aiogram in PyCharm  version 3.12

### DISCUSSION THREAD

**Discussion by Yetto2** [aio-libs/aiohttp#7872] [Q&A]
Collecting aiogram==3.1.1
  Using cached aiogram-3.1.1-py3-none-any.whl.metadata (7.1 kB)
Collecting aiofiles~=23.1.0 (from aiogram==3.1.1)
  Using cached aiofiles-23.1.0-py3-none-any.whl (14 kB)
Collecting aiohttp~=3.8.5 (from aiogram==3.1.1)
  Using cached aiohttp-3.8.6.tar.gz (7.4 MB)
  Installing build dependencies: started
  Installing build dependencies: finished with status 'done'
  Getting requirements to build wheel: started
  Getting requirements to build wheel: finished with status 'done'
  Installing backend dependencies: started
  Installing backend dependencies: finished with status 'done'
  Preparing metadata (pyproject.toml): started
  Preparing metadata (pyproject.toml): finished with status 'done'
Collecting certifi>=2023.7.22 (from aiogram==3.1.1)
  Using cached certifi-2023.11.17-py3-none-any.whl.metadata (2.2 kB)
Collecting magic-filter~=1.0.11 (from aiogram==3.1.1)
  Using cached magic_filter-1.0.12-py3-none-any.whl.metadata (1.5 kB)
Collecting pydantic<2.4,>=2.1.1 (from aiogram==3.1.1)
  Using cached pydantic-2.3.0-py3-none-any.whl.metadata (148 kB)
Collecting typing-extensions~=4.7.1 (from aiogram==3.1.1)
  Using cached typing_extensions-4.7.1-py3-none-any.whl.metadata (3.1 kB)
Requirement already satisfied: attrs>=17.3.0 in c:\users\yetto\pycharmprojects\pythonproject\venv\lib\site-packages (from aiohttp~=3.8.5->aiogram==3.1.1) (23.1.0)
Collecting charset-normalizer<4.0,>=2.0 (from aiohttp~=3.8.5->aiogram==3.1.1)
  Using cached charset_normalizer-3.3.2-cp312-cp312-win_amd64.whl.metadata (34 kB)
Requirement already satisfied: multidict<7.0,>=4.5 in c:\users\yetto\pycharmprojects\pythonproject\venv\lib\site-packages (from aiohttp~=3.8.5->aiogram==3.1.1) (6.0.4)
Collecting async-timeout<5.0,>=4.0.0a3 (from aiohttp~=3.8.5->aiogram==3.1.1)
  Using cached async_timeout-4.0.3-py3-none-any.whl.metadata (4.2 kB)
Requirement already satisfied: yarl<2.0,>=1.0 in c:\users\yetto\pycharmprojects\pythonproject\venv\lib\site-packages (from aiohttp~=3.8.5->aiogram==3.1.1) (1.9.3)
Requirement already satisfied: frozenlist>=1.1.1 in c:\users\yetto\pycharmprojects\pythonproject\venv\lib\site-packages (from aiohttp~=3.8.5->aiogram==3.1.1) (1.4.0)
Requirement already satisfied: aiosignal>=1.1.2 in c:\users\yetto\pycharmprojects\pythonproject\venv\lib\site-packages (from aiohttp~=3.8.5->aiogram==3.1.1) (1.3.1)
Collecting annotated-types>=0.4.0 (from pydantic<2.4,>=2.1.1->aiogram==3.1.1)
  Using cached annotated_types-0.6.0-py3-none-any.whl.metadata (12 kB)
Collecting pydantic-core==2.6.3 (from pydantic<2.4,>=2.1.1->aiogram==3.1.1)
  Using cached pydantic_core-2.6.3-cp312-none-win_amd64.whl.metadata (6.6 kB)
Requirement already satisfied: idna>=2.0 in c:\users\yetto\pycharmprojects\pythonproject\venv\lib\site-packages (from yarl<2.0,>=1.0->aiohttp~=3.8.5->aiogram==3.1.1) (3.4)
Using cached aiogram-3.1.1-py3-none-any.whl (462 kB)
Using cached certifi-2023.11.17-py3-none-any.whl (162 kB)
Using cached magic_filter-1.0.12-py3-none-any.whl (11 kB)
Using cached pydantic-2.3.0-py3-none-any.whl (374 kB)
Using cached pydantic_core-2.6.3-cp312-none-win_amd64.whl (1.8 MB)
Using cached typing_extensions-4.7.1-py3-none-any.whl (33 kB)
Using cached annotated_types-0.6.0-py3-none-any.whl (12 kB)
Using cached async_timeout-4.0.3-py3-none-any.whl (5.7 kB)
Using cached charset_normalizer-3.3.2-cp312-cp312-win_amd64.whl (100 kB)
Building wheels for collected packages: aiohttp
  Building wheel for aiohttp (pyproject.toml): started
  Building wheel for aiohttp (pyproject.toml): finished with status 'error'
Failed to build aiohttp

  error: subprocess-exited-with-error
  
  Building wheel for aiohttp (pyproject.toml) did not run successfully.
  exit code: 1
  
  [110 lines of output]
  *********************
  * Accelerated build *
  *********************
  running bdist_wheel
  running build
  running build_py
  creating build
  creating build\lib.win-amd64-cpython-312
  creating build\lib.win-amd64-cpython-312\aiohttp
  copying aiohttp\abc.py -> build\lib.win-amd64-cpython-312\aiohttp
  copying aiohttp\base_protocol.py -> build\lib.win-amd64-cpython-312\aiohttp
  copying aiohttp\client.py -> build\lib.win-amd64-cpython-312\aiohttp
  copying aiohttp\client_exceptions.py -> build\lib.win-amd64-cpython-312\aiohttp
  copying aiohttp\client_proto.py -> build\lib.win-amd64-cpython-312\aiohttp
  copying aiohttp\client_reqrep.py -> build\lib.win-amd64-cpython-312\aiohttp
  copying aiohttp\client_ws.py -> build\lib.win-amd64-cpython-312\aiohttp
  copying aiohttp\connector.py -> build\lib.win-amd64-cpython-312\aiohttp
  copying aiohttp\cookiejar.py -> build\lib.win-amd64-cpython-312\aiohttp
  copying aiohttp\formdata.py -> build\lib.win-amd64-cpython-312\aiohttp
  copying aiohttp\hdrs.py -> build\lib.win-amd64-cpython-312\aiohttp
  copying aiohttp\helpers.py -> build\lib.win-amd64-cpython-312\aiohttp
  copying aiohttp\http.py -> build\lib.win-amd64-cpython-312\aiohttp
  copying aiohttp\http_exceptions.py -> build\lib.win-amd64-cpython-312\aiohttp
  copying aiohttp\http_parser.py -> build\lib.win-amd64-cpython-312\aiohttp
  copying aiohttp\http_websocket.py -> build\lib.win-amd64-cpython-312\aiohttp
  copying aiohttp\http_writer.py -> build\lib.win-amd64-cpython-312\aiohttp
  copying aiohttp\locks.py -> build\lib.win-amd64-cpython-312\aiohttp
  copying aiohttp\log.py -> build\lib.win-amd64-cpython-312\aiohttp
  copying aiohttp\multipart.py -> build\lib.win-amd64-cpython-312\aiohttp
  copying aiohttp\payload.py -> build\lib.win-amd64-cpython-312\aiohttp
  copying aiohttp\payload_streamer.py -> build\lib.win-amd64-cpython-312\aiohttp
  copying aiohttp\pytest_plugin.py -> build\lib.win-amd64-cpython-312\aiohttp
  copying aiohttp\resolver.py -> build\lib.win-amd64-cpython-312\aiohttp
  copying aiohttp\streams.py -> build\lib.win-amd64-cpython-312\aiohttp
  copying aiohttp\tcp_helpers.py -> build\lib.win-amd64-cpython-312\aiohttp
  copying aiohttp\test_utils.py -> build\lib.win-amd64-cpython-312\aiohttp
  copying aiohttp\tracing.py -> build\lib.win-amd64-cpython-312\aiohttp
  copying aiohttp\typedefs.py -> build\lib.win-amd64-cpython-312\aiohttp
  copying aiohttp\web.py -> build\lib.win-amd64-cpython-312\aiohttp
  copying aiohttp\web_app.py -> build\lib.win-amd64-cpython-312\aiohttp
  copying aiohttp\web_exceptions.py -> build\lib.win-amd64-cpython-312\aiohttp
  copying aiohttp\web_fileresponse.py -> build\lib.win-amd64-cpython-312\aiohttp
  copying aiohttp\web_log.py -> build\lib.win-amd64-cpython-312\aiohttp
  copying aiohttp\web_middlewares.py -> build\lib.win-amd64-cpython-312\aiohttp
  copying aiohttp\web_protocol.py -> build\lib.win-amd64-cpython-312\aiohttp
  copying aiohttp\web_request.py -> build\lib.win-amd64-cpython-312\aiohttp
  copying aiohttp\web_response.py -> build\lib.win-amd64-cpython-312\aiohttp
  copying aiohttp\web_routedef.py -> build\lib.win-amd64-cpython-312\aiohttp
  copying aiohttp\web_runner.py -> build\lib.win-amd64-cpython-312\aiohttp
  copying aiohttp\web_server.py -> build\lib.win-amd64-cpython-312\aiohttp
  copying aiohttp\web_urldispatcher.py -> build\lib.win-amd64-cpython-312\aiohttp
  copying aiohttp\web_ws.py -> build\lib.win-amd64-cpython-312\aiohttp
  copying aiohttp\worker.py -> build\lib.win-amd64-cpython-312\aiohttp
  copying aiohttp\__init__.py -> build\lib.win-amd64-cpython-312\aiohttp
  running egg_info
  writing aiohttp.egg-info\PKG-INFO
  writing dependency_links to aiohttp.egg-info\dependency_links.txt
  writing requirements to aiohttp.egg-info\requires.txt
  writing top-level names to aiohttp.egg-info\top_level.txt
  reading manifest file 'aiohttp.egg-info\SOURCES.txt'
  reading manifest template 'MANIFEST.in'
  warning: no files found matching 'aiohttp' anywhere in distribution
  warning: no previously-included files matching '*.pyc' found anywhere in distribution
  warning: no previously-included files matching '*.pyd' found anywhere in distribution
  warning: no previously-included files matching '*.so' found anywhere in distribution
  warning: no previously-included files matching '*.lib' found anywhere in distribution
  warning: no previously-included files matching '*.dll' found anywhere in distribution
  warning: no previously-included files matching '*.a' found anywhere in distribution
  warning: no previously-included files matching '*.obj' found anywhere in distribution
  warning: no previously-included files found matching 'aiohttp\*.html'
  no previously-included directories found matching 'docs\_build'
  adding license file 'LICENSE.txt'
  writing manifest file 'aiohttp.egg-info\SOURCES.txt'
  copying aiohttp\_cparser.pxd -> build\lib.win-amd64-cpython-312\aiohttp
  copying aiohttp\_find_header.pxd -> build\lib.win-amd64-cpython-312\aiohttp
  copying aiohttp\_headers.pxi -> build\lib.win-amd64-cpython-312\aiohttp
  copying aiohttp\_helpers.pyi -> build\lib.win-amd64-cpython-312\aiohttp
  copying aiohttp\_helpers.pyx -> build\lib.win-amd64-cpython-312\aiohttp
  copying aiohttp\_http_parser.pyx -> build\lib.win-amd64-cpython-312\aiohttp
  copying aiohttp\_http_writer.pyx -> build\lib.win-amd64-cpython-312\aiohttp
  copying aiohttp\_websocket.pyx -> build\lib.win-amd64-cpython-312\aiohttp
  copying aiohttp\py.typed -> build\lib.win-amd64-cpython-312\aiohttp
  creating build\lib.win-amd64-cpython-312\aiohttp\.hash
  copying aiohttp\.hash\_cparser.pxd.hash -> build\lib.win-amd64-cpython-312\aiohttp\.hash
  copying aiohttp\.hash\_find_header.pxd.hash -> build\lib.win-amd64-cpython-312\aiohttp\.hash
  copying aiohttp\.hash\_helpers.pyi.hash -> build\lib.win-amd64-cpython-312\aiohttp\.hash
  copying aiohttp\.hash\_helpers.pyx.hash -> build\lib.win-amd64-cpython-312\aiohttp\.hash
  copying aiohttp\.hash\_http_parser.pyx.hash -> build\lib.win-amd64-cpython-312\aiohttp\.hash
  copying aiohttp\.hash\_http_writer.pyx.hash -> build\lib.win-amd64-cpython-312\aiohttp\.hash
  copying aiohttp\.hash\_websocket.pyx.hash -> build\lib.win-amd64-cpython-312\aiohttp\.hash
  copying aiohttp\.hash\hdrs.py.hash -> build\lib.win-amd64-cpython-312\aiohttp\.hash
  running build_ext
  building 'aiohttp._websocket' extension
  creating build\temp.win-amd64-cpython-312
  creating build\temp.win-amd64-cpython-312\Release
  creating build\temp.win-amd64-cpython-312\Release\aiohttp
  "C:\Program Files (x86)\Microsoft Visual Studio\2022\BuildTools\VC\Tools\MSVC\14.38.33130\bin\HostX86\x64\cl.exe" /c /nologo /O2 /W3 /GL /DNDEBUG /MD -IC:\Users\Yetto\PycharmProjects\pythonProject\venv\include -IC:\Users\Yetto\AppData\Local\Programs\Python\Python312\include -IC:\Users\Yetto\AppData\Local\Programs\Python\Python312\Include "-IC:\Program Files (x86)\Microsoft Visual Studio\2022\BuildTools\VC\Tools\MSVC\14.38.33130\include" "-IC:\Program Files (x86)\Microsoft Visual Studio\2022\BuildTools\VC\Auxiliary\VS\include" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.22621.0\ucrt" "-IC:\Program Files (x86)\Windows Kits\10\\include\10.0.22621.0\\um" "-IC:\Program Files (x86)\Windows Kits\10\\include\10.0.22621.0\\shared" "-IC:\Program Files (x86)\Windows Kits\10\\include\10.0.22621.0\\winrt" "-IC:\Program Files (x86)\Windows Kits\10\\include\10.0.22621.0\\cppwinrt" "-IC:\Program Files (x86)\Windows Kits\NETFXSDK\4.8\include\um" /Tcaiohttp/_websocket.c /Fobuild\temp.win-amd64-cpython-312\Release\aiohttp/_websocket.obj
  _websocket.c
  aiohttp/_websocket.c(1475): warning C4996: 'Py_OptimizeFlag': deprecated in 3.12
  aiohttp/_websocket.c(3042): error C2039: "ob_digit": ­Ґ пў«пҐвбп з«Ґ­®¬ "_longobject".
  C:\Users\Yetto\AppData\Local\Programs\Python\Python312\include\cpython/longintrepr.h(87): note:  б¬. ®Ўкпў«Ґ­ЁҐ "_longobject"
  aiohttp/_websocket.c(3097): error C2039: "ob_digit": ­Ґ пў«пҐвбп з«Ґ­®¬ "_longobject".
  C:\Users\Yetto\AppData\Local\Programs\Python\Python312\include\cpython/longintrepr.h(87): note:  б¬. ®Ўкпў«Ґ­ЁҐ "_longobject"
  aiohttp/_websocket.c(3238): error C2039: "ob_digit": ­Ґ пў«пҐвбп з«Ґ­®¬ "_longobject".
  C:\Users\Yetto\AppData\Local\Programs\Python\Python312\include\cpython/longintrepr.h(87): note:  б¬. ®Ўкпў«Ґ­ЁҐ "_longobject"
  aiohttp/_websocket.c(3293): error C2039: "ob_digit": ­Ґ пў«пҐвбп з«Ґ­®¬ "_longobject".
  C:\Users\Yetto\AppData\Local\Programs\Python\Python312\include\cpython/longintrepr.h(87): note:  б¬. ®Ўкпў«Ґ­ЁҐ "_longobject"
  aiohttp/_websocket.c(3744): error C2039: "ob_digit": ­Ґ пў«пҐвбп з«Ґ­®¬ "_longobject".
  C:\Users\Yetto\AppData\Local\Programs\Python\Python312\include\cpython/longintrepr.h(87): note:  б¬. ®Ўкпў«Ґ­ЁҐ "_longobject"
  error: command 'C:\\Program Files (x86)\\Microsoft Visual Studio\\2022\\BuildTools\\VC\\Tools\\MSVC\\14.38.33130\\bin\\HostX86\\x64\\cl.exe' failed with exit code 2
  [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for aiohttp
ERROR: Could not build wheels for aiohttp, which is required to install pyproject.toml-based projects

**Comment by salko-ua** [comment]
I think you should use a virtual environment like poetry or venv
